snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -76,6 +79,93 @@ def __init__(self, *, output, source, project_options,
             os.path.join('~', 'snap', 'lxd', 'common'))
         os.makedirs(self._lxd_common_dir, exist_ok=True)
 
+    def _get_multipass_remote(self):
+        try:
+            containers = subprocess.check_output([
+                'multipass', 'list']).decode()
+            # Trailing space to disambiguate names starting with snapcraft
+            if 'snapcraft ' not in containers:
```

Comment:
This is flaky. We need a better way, if there is no command like "multipass info snapcraft", we should file a bug to get it.

Reply:
```
multipass info snapcraft-dev
Name:           snapcraft-dev
State:          STOPPED
IPv4:           --
Release:        --
Image hash:     a42eb4561c03 (Ubuntu 16.04 LTS)
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         /home/sergiusens/source/snapcraft  => /home/ubuntu/source/snapcraft
```

Reply:
@kalikiana any input, here?

Reply:
Seems like I missed this comment. Sorry about that. Appreciate the ping!

I'm not using `multipass info` here because that will error. JSON output is being implemented https://github.com/CanonicalLtd/multipass/pull/42 and I'm planning to use that when it's released.

Reply:
As Kyle was also asking for finer-grained exception handling, I changed it now to rely on the command failing rather than checking the list. Note that this does produce `instance "snapcraft" does not exist` as a side-effect.